celer/tests/test_enet.py

```diff
@@ -0,0 +1,187 @@
+        X, y = build_dataset(n_samples=30, n_features=50)
+        y = np.sign(y)
+        celer_path(X, y, 'logreg', l1_ratio=0.5)
+
+
+@pytest.mark.parametrize("sparse_X", [True, False])
+def test_ElasticNet_Lasso_equivalence(sparse_X):
+    n_samples, n_features = 30, 50
+    X, y = build_dataset(n_samples, n_features, sparse_X=sparse_X)
+    alpha_max = norm(X.T@y, ord=np.inf) / n_samples
+
+    alpha = alpha_max / 100.
+    coef_lasso = Lasso(alpha=alpha).fit(X, y).coef_
+    coef_enet = ElasticNet(alpha=alpha, l1_ratio=1.0).fit(X, y).coef_
+
+    assert_allclose(coef_lasso, coef_enet)
+
+    np.random.seed(0)
+    weights = abs(np.random.randn(n_features))
+    alpha_max = norm(X.T@y / weights, ord=np.inf) / n_samples
+
+    alpha = alpha_max / 100.
+    coef_lasso = Lasso(alpha=alpha, weights=weights).fit(X, y).coef_
+    coef_enet = ElasticNet(alpha=alpha, l1_ratio=1.0, weights=weights).fit(X, y).coef_
+
+    assert_allclose(coef_lasso, coef_enet)
+
+
+@pytest.mark.parametrize("sparse_X, prune", [(False, 0), (False, 1)])
+def test_celer_enet_sk_enet_equivalence(sparse_X, prune):
+    """Test that celer_path matches sklearn enet_path."""
+
+    n_samples, n_features = 30, 50
+    X, y = build_dataset(n_samples, n_features, sparse_X=sparse_X)
+
+    tol = 1e-14
+    l1_ratio = 0.7
+    alpha_max = norm(X.T@y, ord=np.inf) / n_samples
+    params = dict(eps=1e-3, alphas=[alpha_max / 100.], tol=tol, l1_ratio=l1_ratio)
+
+    alphas1, coefs1, gaps1 = celer_path(
+        X, y, "lasso", return_thetas=False, verbose=0, prune=prune,
+        max_iter=30, **params)
+
+    alphas2, coefs2, _ = enet_path(X, y, verbose=0, **params,
+                                   max_iter=10000)
+
+    assert_equal(alphas1, alphas2)
+    assert_array_less(gaps1, tol * norm(y) ** 2 / n_samples)
+    assert_allclose(coefs1, coefs2, rtol=1e-3, atol=1e-4)
+
+
+@pytest.mark.parametrize("sparse_X, prune", product([False], [0, 1]))
+def test_celer_enet_sk_enet_equivalence_many(sparse_X, prune):
+    """Test that celer_path matches sklearn enet_path."""
+
+    n_samples, n_features = 30, 50
+    X, y = build_dataset(n_samples, n_features, sparse_X=sparse_X)
+
+    tol = 1e-14
+    l1_ratio = 0.7
+    n_alphas = 20
+    params = dict(eps=1e-3, n_alphas=n_alphas, tol=tol, l1_ratio=l1_ratio)
+
+    alphas1, coefs1, gaps1 = celer_path(
+        X, y, "lasso", return_thetas=False, verbose=0, prune=prune,
+        max_iter=30, **params)
+
+    alphas2, coefs2, _ = enet_path(X, y, verbose=0, **params,
+                                   max_iter=10000)
+
+    assert_allclose(alphas1, alphas2)
+    assert_array_less(gaps1, tol * norm(y) ** 2 / n_samples)
+    assert_allclose(coefs1, coefs2, rtol=1e-3, atol=1e-4)
+
+
+@pytest.mark.parametrize("sparse_X, fit_intercept, positive",
+                         product([False], [False, True], [False, False]))
+def test_celer_ElasticNet_vs_sk_ElasticNet(sparse_X, fit_intercept, positive):
```

Comment:
use shorter names. you had the good idea to call the file `test_enet` so the test can be names `test_celer_sklearn`